src/jgikbase/idmapping/config.py

```diff
@@ -0,0 +1,140 @@
+"""
+Configuration handlers for the ID Mapping system.
+"""
+from typing import Dict, Optional
+from pathlib import Path
+import os
+import configparser
+# May want different configuration implementations based on the deployment environment.
+# YAGNI for now.
+
+# TODO CFG support user handler custom configs
+# TODO CFG support which user handlers are enabled
+# TODO CFG support which user handlers are allowed to specify admins
+
+
+class IDMappingConfigError(Exception):
+    """ Thrown when there's an error in the ID Mapping system configuration. """
+    pass
+
+
+class KBaseConfig:
+    """
+    Loads a configuration from a standard KBase-style deploy.cfg file (an ini file with only
+    one section.) The configuration is contained in the `idmapping` section of the config file.
+
+    The keys are:
+    mongo-host
+    mongo-db
+    mongo-user
+    mongo-pwd
+    dont-trust-x-ip-headers
+
+    The last key is optional and instructs the server to ignore the X-Real-IP and X-Forwarded-For
+    headers if set to the string 'true'.
+
+    :ivar mongo_host: the host of the MongoDB instance, including the port.
+    :ivar mongo_db: the MongoDB database to use for the ID mapping service.
+    :ivar mongo_user: the username to use with MongoDB, if any.
+    :ivar mongo_pwd: the password to use with MongoDB, if any.
+    :ivar ignore_ip_headers: True if the X-Real-IP and X-Forwarded-For headers should be ignored.
+    """
+
+    ENV_VAR_IDMAPPING = 'ID_MAPPING_CONFIG'
+    """
```

Comment:
Just out of curiosity, do docstrings places after these class vars show up in __doc__, or are they used by sphinx?

Reply:
They show up in sphinx; I've never used `doc` so I can't answer that one.